src/contexts/Accounts/Proxies/index.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useEffect, useRef, useState } from 'react';
+import type { AnyApi, MaybeAccount } from 'types';
+import { clipAddress, rmCommas, setStateWithRef } from 'Utils';
+import * as defaults from './defaults';
+import type {
+  Delegate,
+  ProxiesContextInterface,
+  Proxy,
+  ProxyAccount,
+} from './type';
+
+export const ProxiesContext = React.createContext<ProxiesContextInterface>(
+  defaults.defaultProxiesContext
+);
+
+export const useProxies = () => React.useContext(ProxiesContext);
+
+export const ProxiesProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const { api, isReady, network } = useApi();
+  const { accounts, activeAccount } = useConnect();
+
+  // store the proxy accounts of each imported account.
+  const [proxies, setProxies] = useState<Array<Proxy>>([]);
+  const proxiesRef = useRef(proxies);
+
+  const [unsubs, setUnsubs] = useState<AnyApi>([]);
+  const unsubsRef = useRef(unsubs);
+
+  const [delegators, setDelegators] = useState<Array<MaybeAccount>>([]);
+  const delegatorsRef = useRef(delegators);
+
+  useEffect(() => {
+    if (isReady) {
+      // local updated values
+      let newProxies = proxiesRef.current;
+      const newUnsubsProxy = unsubsRef.current;
+
+      // get accounts removed: use these to unsubscribe
+      const accountsRemoved = proxiesRef.current.filter(
+        (a: Proxy) =>
+          !accounts.find((c: ImportedAccount) => c.address === a.delegator)
+      );
+
+      const accountsAdded = accounts.filter(
+        (c: ImportedAccount) =>
+          !proxiesRef.current.find((a: Proxy) => a.delegator === c.address)
+      );
+
+      // update proxy state for removal
+      newProxies = proxiesRef.current.filter((l: Proxy) =>
+        accounts.find((c: ImportedAccount) => c.address === l.delegator)
+      );
+      if (newProxies.length < proxiesRef.current.length) {
+        accountsRemoved.forEach((p: Proxy) => {
+          const unsub = unsubsRef.current.find(
+            (u: AnyApi) => u.key === p.delegator
+          );
+          if (unsub) {
+            unsub.unsub();
+            newUnsubsProxy.filter((u: AnyApi) => u.key !== p.delegator);
+          }
+        });
+        setStateWithRef(newProxies, setProxies, proxiesRef);
+        setStateWithRef(newUnsubsProxy, setUnsubs, unsubsRef);
+      }
+
+      // if accounts have changed, update state with new unsubs / accounts
+      if (accountsAdded.length) {
+        // subscribe to added accounts
+        accountsAdded.map((a: ImportedAccount) =>
+          subscribeToProxies(a.address)
+        );
+      }
+    }
+  }, [accounts, isReady, network]);
+
+  // unsubscribe from proxy subscriptions on unmount
+  useEffect(() => {
+    Object.values(unsubsRef.current).forEach(({ unsub }: AnyApi) => {
+      unsub();
+    });
+  }, []);
+
+  useEffect(() => {
+    const allProxies = [];
+    const proxiedAccounts = [];
+    for (const a of accounts) {
```

Comment:
Why is `accounts` being used? This `useEffect` just needs to loop through `proxiesRef.current` and reverse the record to be `delegate -> proxies` instead.